05week/checkers.js

```diff
@@ -2,111 +2,305 @@
+        // checks to see if the checker moved diagionally 1 space
+        if ((OR - 1 === DR && OC - 1 === DC) ||
+            (OR - 1 === DR && OC + 1 === DC) ||
+            (OR + 1 === DR && OC - 1 === DC) ||
+            (OR + 1 === DR && OC + 1 === DC)) {
+            return true;
+        } else if ((OR - 2 === DR && OC - 2 === DC) ||
+            (OR - 2 === DR && OC + 2 === DC) ||
+            (OR + 2 === DR && OC - 2 === DC) ||
+            (OR + 2 === DR && OC + 2 === DC)) {
+            return true;
+        } else {
+            console.log("!!ERROR: Invalid move");
+            return false;
+        }
+
+    }
+
+    this.removeChecker = function(inputs) {
+        // sets origin to null
+        this.board.grid[inputs.originRow][inputs.originColumn] = null;
+        return;
+
+    }
+
+    this.addChecker = function(inputs) {
+        // adds a checker piece to destination
+        this.board.grid[inputs.destinationRow][inputs.destinationColumn] = new Checker(this.playerTurn);
+        return;
     }
-  };
-
-  // prints out the board
-  this.viewGrid = function() {
-    // add our column numbers
-    let string = "  0 1 2 3 4 5 6 7\n";
-    for (let row = 0; row < 8; row++) {
-      // we start with our row number in our array
-      const rowOfCheckers = [row];
-      // a loop within a loop
-      for (let column = 0; column < 8; column++) {
-        // if the location is "truthy" (contains a checker piece, in this case)
-        if (this.grid[row][column]) {
-          // push the symbol of the check in that location into the array
-          rowOfCheckers.push(this.grid[row][column].symbol);
+
+    this.togglePlayer = function() {
+        // switch to the next player
+        if (this.playerTurn === 'R') {
+            this.playerTurn = 'B';
         } else {
-          // just push in a blank space
-          rowOfCheckers.push(' ');
-        }
-      }
-      // join the rowOfCheckers array to a string, separated by a space
-      string += rowOfCheckers.join(' ');
-      // add a 'new line'
-      string += "\n";
+            this.playerTurn = 'R';
+        }
     }
-    console.log(string);
-  };
 
-  // Your code here
-}
-function Game() {
+    this.removeOpponent = function(inputs) {
+
+        // easy to read
+        let OR = inputs.originRow;
+        let OC = inputs.originColumn;
+        let DR = inputs.destinationRow;
+        let DC = inputs.destinationColumn;
 
-  this.board = new Board();
+        if (Math.abs(DR - OR) === 2) {
+            let killRow = DR - OR > 0 ? OR + 1 : DR + 1;
+            let killCol = DR - OC > 0 ? OC + 1 : DR + 1;
+            console.log('!! A checker was captured !!');
+            this.board.grid[killRow][killCol] = null;
+        }
+    }
+
+    this.moveChecker = function(origin, destination) {
+
+        // checks to see if the inputs are valid
+        let inputs = new this.ValidateInputs(origin, destination); // is there a better way to do this as to avoid passing in board?
```

Comment:
If you are making an instance by calling ```new this.ValidateInputs(origin, destination)```, that returns a valid Input instance, i would expect the class name to be a noun and not an action.